disttask/importinto/subtask_executor.go

```diff
@@ -73,6 +84,140 @@ func (e *postProcessMinimalTaskExecutor) Run(ctx context.Context) error {
 	return postProcess(ctx, mTask.taskMeta, &mTask.meta, mTask.logger)
 }
 
+// postProcess does the post-processing for the task.
+func postProcess(ctx context.Context, taskMeta *TaskMeta, subtaskMeta *PostProcessStepMeta, logger *zap.Logger) (err error) {
+	failpoint.Inject("syncBeforePostProcess", func() {
+		TestSyncChan <- struct{}{}
+		<-TestSyncChan
+	})
+
+	logger.Info("post process")
+
+	// TODO: create table indexes depends on the option.
+	// create table indexes even if the post process is failed.
+	// defer func() {
+	// 	err2 := createTableIndexes(ctx, globalTaskManager, taskMeta, logger)
+	// 	err = multierr.Append(err, err2)
+	// }()
+
+	return verifyChecksum(ctx, taskMeta, subtaskMeta, logger)
+}
+
+func verifyChecksum(ctx context.Context, taskMeta *TaskMeta, subtaskMeta *PostProcessStepMeta, logger *zap.Logger) error {
+	if taskMeta.Plan.Checksum == config.OpLevelOff {
+		return nil
+	}
+	localChecksum := verify.MakeKVChecksum(subtaskMeta.Checksum.Size, subtaskMeta.Checksum.KVs, subtaskMeta.Checksum.Sum)
+	logger.Info("local checksum", zap.Object("checksum", &localChecksum))
+
+	failpoint.Inject("waitCtxDone", func() {
+		<-ctx.Done()
+	})
+
+	globalTaskManager, err := storage.GetTaskManager()
+	if err != nil {
+		return err
+	}
+	remoteChecksum, err := checksumTable(ctx, globalTaskManager, taskMeta, logger)
+	if err != nil {
+		return err
+	}
+	if !remoteChecksum.IsEqual(&localChecksum) {
+		err2 := common.ErrChecksumMismatch.GenWithStackByArgs(
+			remoteChecksum.Checksum, localChecksum.Sum(),
+			remoteChecksum.TotalKVs, localChecksum.SumKVS(),
+			remoteChecksum.TotalBytes, localChecksum.SumSize(),
+		)
+		if taskMeta.Plan.Checksum == config.OpLevelOptional {
+			logger.Warn("verify checksum failed, but checksum is optional, will skip it", zap.Error(err2))
+			err2 = nil
+		}
+		return err2
+	}
+	logger.Info("checksum pass", zap.Object("local", &localChecksum))
+	return nil
+}
+
+func checksumTable(ctx context.Context, executor storage.SessionExecutor, taskMeta *TaskMeta, logger *zap.Logger) (*local.RemoteChecksum, error) {
+	var (
+		tableName                    = common.UniqueTable(taskMeta.Plan.DBName, taskMeta.Plan.TableInfo.Name.L)
+		sql                          = "ADMIN CHECKSUM TABLE " + tableName
+		maxErrorRetryCount           = 3
+		distSQLScanConcurrencyFactor = 1
+		remoteChecksum               *local.RemoteChecksum
+		txnErr                       error
+	)
+
+	ctx = util.WithInternalSourceType(ctx, kv.InternalImportInto)
+	for i := 0; i < maxErrorRetryCount; i++ {
+		txnErr = executor.WithNewTxn(ctx, func(se sessionctx.Context) error {
+			// increase backoff weight
+			if err := setBackoffWeight(se, taskMeta, logger); err != nil {
+				logger.Warn("set tidb_backoff_weight failed", zap.Error(err))
+			}
+
+			// reduce distsql scan concurrency
```

Comment:
the comment should also explain the reason. This behaviour is very plain, no need to repeat it.